pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -180,6 +231,118 @@ func createRawWorkerDeployment(componentMeta metav1.ObjectMeta,
 	return deployment
 }
 
+func GetKServeContainerPort(podSpec *corev1.PodSpec) string {
+	for _, container := range podSpec.Containers {
+		if container.Name == "kserve-container" {
+			if len(container.Ports) > 0 {
+				return strconv.Itoa(int(container.Ports[0].ContainerPort))
+			}
+		}
+	}
+	return ""
+}
+func generateOauthProxyContainer(clientset kubernetes.Interface, isvc string, namespace string, upstreamPort string) (corev1.Container, error) {
+	configMap, err := clientset.CoreV1().ConfigMaps(constants.KServeNamespace).Get(context.TODO(), constants.InferenceServiceConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		return corev1.Container{}, err
+	}
+	oauthProxyJSON := strings.TrimSpace(configMap.Data["oauthProxy"])
+	oauthProxyConfig := v1beta1.OauthConfig{}
+	if err := json.Unmarshal([]byte(oauthProxyJSON), &oauthProxyConfig); err != nil {
+		return corev1.Container{}, err
+	}
+	if oauthProxyConfig.Image == "" || oauthProxyConfig.MemoryRequest == "" || oauthProxyConfig.MemoryLimit == "" ||
+		oauthProxyConfig.CpuRequest == "" || oauthProxyConfig.CpuLimit == "" {
+		return corev1.Container{}, fmt.Errorf("one or more oauthProxyConfig fields are empty")
+	}
+	oauthImage := oauthProxyConfig.Image
+	oauthMemoryRequest := oauthProxyConfig.MemoryRequest
+	oauthMemoryLimit := oauthProxyConfig.MemoryLimit
+	oauthCpuRequest := oauthProxyConfig.CpuRequest
+	oauthCpuLimit := oauthProxyConfig.CpuLimit
+
+	cookieSecret, err := generateCookieSecret()
+	if err != nil {
+		return corev1.Container{}, err
+	}
+
+	return corev1.Container{
+		Name: "oauth-proxy",
+		Args: []string{
```

Comment:
shouldn't we have a service account here?

Reply:
something like:

        - --openshift-service-account=oauth-proxy
        - --client-id=system:serviceaccount:my-namespace:oauth-proxy
        - --client-secret-file=/var/run/secrets/kubernetes.io/serviceaccount/token

Reply:
also, what about the cookie expiration time?

Reply:
@israel-hdez can't the service account be configurable at the time things are deployed?

Reply:
Yes, there is a field in the ISVC that allows the user to configure the SA. KServe should already be covering that.

Reply:
will kserve also attach the SA here?

Reply:
Not here. We can't  have a SA per container. It is per pod.